src/hotspot/share/prims/jvmti.xml

```diff
@@ -11974,7 +11980,8 @@ myInit() {
       There are no Java programming language or JNI stack frames at the specified depth.
     </errorid>
     <errorid id="JVMTI_ERROR_OPAQUE_FRAME" num="32">
-      Information about the frame is not available (e.g. for native frames).
+      Information about the frame is not available (e.g. for native frames),
+      or the frame is not suitable for the requested operation.
```

Comment:
After re-reading the spec changes, I'm wondering if we can improve on "or the frame is not suitable for the requested operation".  StopThread doesn't have a frame parameter. ForceEarlyReturn doesn't have a frame parameter either as it's implicit (the current frame). I wonder if wording something like this might be better:
"or a function on a thread cannot be performed at the thread's current frame".

Reply:
The wording starts off with "Information about the frame...", and you haven't suggested to change that to "the current frame". We should be consistent. Can't we just change both "the frame" references to "the current frame", and leave the rest the same as what Serguei has in place here?

Reply:
I think the first part is okay because it's for functions that are about frames. The NotifyFramePop specifies the depth so it may not be the current frame. The second usage is the functions on a thread where we might do better than "not suitable".

Reply:
Ok. How about "the function cannot be performed on the thread's current frame". We already have a couple references to "the function" in the error codes section.

Reply:
We have two suggestions:
> - "or a function on a thread cannot be performed at the thread's current frame".
> - "the function cannot be performed on the thread's current frame."
So, we need to pick one. The second one looks simpler to me but
I'm not completely sure that it reflects the full meaning correctly.
I wonder about a mix of the two suggestions above:
> - "the function cannot be performed at the thread's current frame."

Reply:
We need to account for the `SetLocalXXX` functions with the `depth` parameter which also return `OPAQUE_FRAME` error code for virtual frames. My concern is if the "current frame" part is fully correct.

Reply:
I've pushed variant from Chris which is a rephrase of what Alan suggested.

Reply:
I can't help thinking we can do better than "on the thread's current frame" but in the absence of a better suggestion then I think what you have is okay.

Reply:
Thank you, Alan.
Updated the CSR.